src/php/ext/grpc/call.c

```diff
@@ -125,15 +125,9 @@ zval *grpc_parse_metadata_array(grpc_metadata_array *metadata_array) {
     memcpy(str_val, elem->value, elem->value_length);
     if (zend_hash_find(array_hash, str_key, key_len, (void **)data) ==
         SUCCESS) {
-      if (Z_TYPE_P(*data) != IS_ARRAY) {
-        zend_throw_exception(zend_exception_get_default(),
-                             "Metadata hash somehow contains wrong types.",
-                             1 TSRMLS_CC);
-          efree(str_key);
-          efree(str_val);
```

Comment:
And if the condition is unreachable, just remove it. Otherwise, handle it properly. But don't just remove part of the handling code.